modin/pandas/series.py

```diff
@@ -161,20 +161,31 @@ def _getitem(self, key):
             # each item individually.
             key = key._to_pandas()
         if is_bool_indexer(key):
-            return self.loc[self.index[key]]
+            return self.__constructor__(
```

Comment:
Can we change `base.py::loc()` or the indexing class instead of having the change there?

Reply:
I don't follow what you mean here.

Reply:
Instead of calling `getitem_row_array`, can we change `loc()` to do all of that for us?

Reply:
`loc` creates a `LocIndexer` object. Eventually we do need to merge the codepaths, but this PR is about performance and that would be a good first issue.